modin/core/dataframe/pandas/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,455 @@
+            null = ColumnNullType.NON_NULLABLE
+        elif kind == DTypeKind.CATEGORICAL:
+            # Null values for categoricals are stored as `-1` sentinel values
+            # in the category date (e.g., `col.values.codes` is int8 np.ndarray)
+            null = ColumnNullType.USE_SENTINEL
+            value = -1
+        elif kind == DTypeKind.STRING:
+            null = ColumnNullType.USE_BYTEMASK
+            value = (
+                0  # follow Arrow in using 1 as valid value and 0 for missing/null value
+            )
+        else:
+            raise NotImplementedError(f"Data type {kind} not yet supported")
+
+        return null, value
+
+    _null_count_cache = None
+
+    # TODO: since python 3.9:
+    # @cached_property
+    @property
+    def null_count(self) -> int:
+        if self._null_count_cache is not None:
+            return self._null_count_cache
+
+        def map_func(df):
+            return df.isna()
+
+        def reduce_func(df):
+            return pandas.DataFrame(df.sum())
+
+        intermediate_df = self._col.tree_reduce(0, map_func, reduce_func)
+        # Set ``pandas.RangeIndex(1)`` to index and column labels because
+        # 1) We internally use '__reduced__' for labels of a reduced axis
+        # 2) The return value of `reduce_func` is a pandas DataFrame with
+        # index and column labels set to ``pandas.RangeIndex(1)``
+        # 3) We further use `to_pandas().squeeze()` to get an integer value of the null count
+        intermediate_df.index = pandas.RangeIndex(1)
+        intermediate_df.columns = pandas.RangeIndex(1)
+        self._null_count_cache = intermediate_df.to_pandas().squeeze()
+        return self._null_count_cache
+
+    @property
+    def metadata(self) -> Dict[str, Any]:
+        return {"modin.index": self._col.index}
+
+    def num_chunks(self) -> int:
+        return self._col._partitions.shape[0]
+
+    def get_chunks(
+        self, n_chunks: Optional[int] = None
+    ) -> Iterable["PandasProtocolColumn"]:
+        cur_n_chunks = self.num_chunks()
+        n_rows = self.size
+        if n_chunks is None or n_chunks == cur_n_chunks:
+            cum_row_lengths = np.cumsum([0] + self._col._row_lengths)
+            for i in range(len(cum_row_lengths) - 1):
+                yield PandasProtocolColumn(
+                    self._col.mask(
+                        row_positions=range(cum_row_lengths[i], cum_row_lengths[i + 1]),
+                        col_positions=None,
+                    ),
+                    allow_copy=self._col._allow_copy,
+                )
+            return
+
+        if n_chunks % cur_n_chunks != 0:
+            raise RuntimeError(
+                "The passed `n_chunks` must be a multiple of `self.num_chunks()`."
+            )
+
+        if n_chunks > n_rows:
+            raise RuntimeError(
+                "The passed `n_chunks` value is bigger than `self.num_rows()`."
+            )
```

Comment:
Yes, I think it is an error because `n_chunks` must be a multiple of `self.num_chunks()` meaning the producer must subdivide each chunk before yielding it, but not vice versa.